api/v1beta1/machine_types.go

```diff
@@ -230,10 +230,22 @@ type MachineStatus struct {
 	// Conditions defines current service state of the Machine.
 	// +optional
 	Conditions Conditions `json:"conditions,omitempty"`
+
+	// Deletion is the deletion state of the Machine.
+	// +optional
+	Deletion MachineStatusDeletion `json:"deletion,omitempty"`
 }
 
 // ANCHOR_END: MachineStatus
 
+// MachineStatusDeletion is the deletion state of the Machine.
+type MachineStatusDeletion struct {
+	// NodeDrainStartTime is the time when the drain of the node started.
+	NodeDrainStartTime *metav1.Time `json:"nodeDrainStartTime,omitempty"`
```

Comment:
Do we also need a finish time? Same for the other field?

Reply:
I personally don't think that drain or detach finish time is an important info to have in the API (users & SRE mostly care about what is going on now and eventually why it is stuck, rarely they care about at what happened and for this log a more exhaustive)
But no strong opinion.

Reply:
> how do I know that they have finished if I don't have some field to tell me? 🤔

From controller-perspective we (at least currently) do not care:
* either the controller tries to drain again which should be a no-op (happy path)
* or the drain is skipped because the timeout is reached.

From the user perspective: the information where the deletion is at should be part of the `Deleting` condition I'd say.

Reply:
I think in general the new Deleting condition should make clear at which phase of the deletion workflow we are (including making clear which parts are already completed)